### PR TITLE
cast sendto to SendError inside send

### LINKED ISSUE #7474: SendError and SendToError conflicts in the definition of `send` after #7417
zig version 6dfe9cc83ea75a663b4f6dfa488b95950207a1cf

```
date/master λ zig build
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4926:18: error: expected type 'std.os.SendError!usize', found 'std.os.SendToError!usize'
    return sendto(sockfd, buf, flags, null, 0);
                 ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4926:18: note: error set 'std.os.SendToError' cannot cast into error set 'std.os.SendError'
    return sendto(sockfd, buf, flags, null, 0);
                 ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4781:5: note: 'error.AddressFamilyNotSupported' not a member of destination error set
    AddressFamilyNotSupported,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4784:5: note: 'error.SymLinkLoop' not a member of destination error set
    SymLinkLoop,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4787:5: note: 'error.NameTooLong' not a member of destination error set
    NameTooLong,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4790:5: note: 'error.FileNotFound' not a member of destination error set
    FileNotFound,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4791:5: note: 'error.NotDir' not a member of destination error set
    NotDir,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4794:5: note: 'error.NetworkUnreachable' not a member of destination error set
    NetworkUnreachable,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4800:5: note: 'error.SocketNotConnected' not a member of destination error set
    SocketNotConnected,
    ^
/gnu/store/a43966k9w3dxmxadwnijjxahwg1z4hm4-zig-master-0.7.1+6dfe9cc/lib/zig/std/os.zig:4802:5: note: 'error.AddressNotAvailable' not a member of destination error set
    AddressNotAvailable,
    ^
```

### PR DESCRIPTION
**Pull request by lun-4**
Is this the right approach...?

Fix #7474.

### CODE REVIEW

File: lib/std/os.zig [hunk 1]
```diff
@@ -4923,7 +4923,10 @@ pub fn send(
     buf: []const u8,
     flags: u32,
 ) SendError!usize {
-    return sendto(sockfd, buf, flags, null, 0);
+    return sendto(sockfd, buf, flags, null, 0) catch |err| switch (err) {
+        error.AddressFamilyNotSupported, error.SymLinkLoop, error.NameTooLong, error.FileNotFound, error.NotDir, error.NetworkUnreachable, error.SystemResources, error.SocketNotConnected, error.WouldBlock, error.AddressNotAvailable => unreachable,
```
Comment (daurnimator) [inline lib/std/os.zig:4927]: each error should be on its own line.

Not all of these are unreachable. e.g. `WouldBlock` is possible.

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: lib/std/os.zig (hunk 1 above)
Comment (lun-4) [reply lib/std/os.zig]: `WouldBlock` was accidentally duplicated. Thanks!

File: lib/std/os.zig [hunk 2]
```diff
@@ -4923,7 +4924,16 @@ pub fn send(
...
-    return sendto(sockfd, buf, flags, null, 0);
+    return sendto(sockfd, buf, flags, null, 0) catch |err| switch (err) {
+        error.AddressFamilyNotSupported => unreachable,
+        error.SymLinkLoop => unreachable,
+        error.NameTooLong => unreachable,
+        error.FileNotFound => unreachable,
+        error.NotDir => unreachable,
+        error.NetworkUnreachable => unreachable,
+        error.AddressNotAvailable => unreachable,
+        else => return @errSetCast(SendError, err),
```
Comment (andrewrk) [inline lib/std/os.zig:4935]: Check this out:

```suggestion
        else => |e| return e,
```

Pretty swanky, right?

**Review by andrewrk — CHANGES_REQUESTED** [review]
With this one change I think this is good

**Pull request merged**